### ISSUE TITLE
rpi_camera 'Image not available'

### ISSUE THREAD

**Issue by robmarkcole** [home-assistant/hassio#184]
I'm seeing 'Image not available' under 0.53 on Pi-zero-W and pi-3 with `platform: rpi_camera`

![image](https://user-images.githubusercontent.com/11855322/30260262-388516a2-96be-11e7-9c47-093ef40d8cf3.png)

**Comment by pvizeli** [comment]
Maybe it use the RT/XT like zwave module and GPIO. Try that: https://home-assistant.io/hassio/zwave/

**Comment by pvizeli** [comment]
Try: `curl -d '{"devices": ["ttyAMA0"]}' http://hassio/homeassistant/options` to map the GPIO interface into homeassistant container

**Comment by pvizeli** [comment]
I think, if we solve this with that command, I add a autodetect for `ttyAMA0` device and remove this command

**Comment by pvizeli** [comment]
You need reboot the host or use the new function of hassio 0.61 to stop/start the homeassistant container with ssh

**Comment by pvizeli** [comment]
Try also first to update config.txt:
![image](https://user-images.githubusercontent.com/15338540/30282550-e8b3d444-9715-11e7-8ac8-378c6f8b95a7.png)

**Comment by robmarkcole** [comment]
Hi, where is config.txt? I have samba share setup as below.
Cheers

![image](https://user-images.githubusercontent.com/11855322/30288497-db21f2e2-9720-11e7-8360-f77705c8a053.png)

**Comment by pvizeli** [comment]
Yeah you need do that on SD card. But @gollo have try it and mean that will not work

**Comment by gollo** [comment]
I'm looking into it when I have the time...

**Comment by gsbrown** [comment]
Same issue, hoping for a fix but have a workaround with a dropcam for now

**Comment by pvizeli** [comment]
The problem look like with a blob firmware that need to be load and docker/alpine. But It is hard to discovery that without hardware :(

**Comment by gsbrown** [comment]
Anyone figure this out yet?

**Comment by jasiekzar** [comment]
I have the same problem - I have enabled the camera in the config file on the sd card but still "image not available"...

**Comment by thesplitter** [comment]
I have tried editing config.txt to enable the camera and gpumem to 128 (as required) but the result is the same

**Comment by gollo** [comment]
The root problem is that the module does not run in the Alpine docker
container.  Ideally, if I get time, I will attempt to get it to  run in
native Alpine on the Pi, then move to attempting to get it to run in the
container.  Attempting to get it running in the container has not worked
for me so far (I know the cam works fine with a raspbian image, native or
in a container, for example).  If anyone has more time that is where I
would focus first.

**Comment by sardauker** [comment]
Hi.
Raspberry pi 3,
Home Assistant 0.57.2,

Same error, I don't think I can even see the SD :/

**Comment by gsbrown** [comment]
Just posting another friendly request for this to be fixed - pi camera is hooked up in my garage (where my pi3 running hassio is located) and it gives me visual confirmation of garage door status. 

I know it is tough to find time to work on a free project so please don't construe this as any form of complaint. Home Assistant is amazing and I just don't want pi camera support in hassio to be forgotten about. 

Thanks to the community for all the work they put into HA!

**Comment by kotopoulos10** [comment]
I'm having the same issue as well HA 0.58.1 where the front end says "Image not available".s Not trying to complain at all either just wondering whats going on and if there's any way to help out! Is there a certain file to look at on the SD card to modify?

**Comment by gollo** [comment]
Please view the entire issue, it has all the details...

**Comment by gdampf** [comment]
Hi gollo,
It's nice to know the cause - but still, most of the users of hass.io have choosen it, because, they don't want to start development themselves - they just want to use the system. And that's why it does not help just to know why it is actually not working. This bug will force them to switch to hassbian or similar. Is anyone still working on that issue or is hass.io not supporting rpi cameras anymore? Or did I miss a sufficient workaround?
Don't missunderstand my impatience, I really appreciate the work the comunity has done so far.
BR
Guido

**Comment by Mryck** [comment]
Hi, I found some information to get the camera module working on resinOS
https://docs.resin.io/hardware/i2c-and-spi/#raspberry-pi-camera-module
In addation to:
```
gpu_mem=128
start_x=1
```
in the config.txt file, modprobe bcm2835-v4l2 is needed before starting scripts in either the package.json start command or Dockerfile CMD command.

**Comment by frenck** [comment]
I've just sent out an order for a couple of camera's.
Will look into this as soon as they arrive.

**Comment by frenck** [comment]
Package has arrived 🎉

**Comment by gollo** [comment]
To update, I tried to make this work on a raw alpine install on the pi and could not, this I believe is the first step to understanding how it might work in the hass.io docker image

**Comment by frenck** [comment]
@gollo I would like to take the approach of trying to make it work inside any Docker container on the Hass.io/ResinOS Docker host.

**Comment by GregoryDosh** [comment]
I was having this same issue and I debugged and seemingly fixed it through noticing that the `rpi_camera.py` component was essentially running the `raspistill` command.  I tried to emulate the command it was running using
```
sudo -u homeassistant raspistill --nopreview -o /tmp/raspistill.jpg -t 0 -w 640 -h 480 -tl 1000 -q 7 -rot 0
```
Which gave me an error:
```
* failed to open vchiq instance
```
And through some googlefu it seems like the homeassistant profile I was running as needed to be added to the raspberry pi video group.
```
sudo adduser homeassistant video
```
Then I restarted the home assistant service
```
sudo systemctl restart home-assistant
```
And it started working just fine.  Hopefully this is enough info for others to figure out what might be wrong with their camera too.

**Comment by doscas** [comment]
Hello all,

I am having the same problem with the RPi camera.
I installed Hassio and enabled SSH.
When I login as root, I cannot run the `sudo adduser homeassistant video` as indicated above.
It says `-bash: sudo: command not found` 

I'm not a pro in programming or anything like it.
Can anyone point me in the right direction? I've been searching for this for so long, I am actually getting frustrated something that should be so easy has become such a pain.
Thank you for any help you can give me.

**Comment by GregoryDosh** [comment]
@doscas If you're already logging in as root you should be able to remove the `sudo` from all of the commands above.

**Comment by doscas** [comment]
Thank you for your comment,@GregoryDosh. 

That's none of the few things I learned.
But when I do, it tells me `addgroup: unknown user homeassistant`
So my question now is: what's the user Home Assistant uses?

**Comment by jasiekzar** [comment]
I have the same - no home assistant user...

**Comment by doscas** [comment]
Has anyone got the PI Camera to work?
It changed to rpi_cam and now back to rpi_camera.
I believe some thing must have been done.
Anyone with news?

**Comment by enurist** [comment]
I just switched from plain raspbian to hassio (instead of hassbian) and I am quite astonished that rpi_camera is not working. Can I be of any help to solve this issue?

**Comment by Vincent-Ma** [comment]
On Hassio there is a workaround that you can do in order to use the Raspberry camera until this issue is fixed:

As @Mryck pointed out in order to use the camera, 2 additional steps need to be done:

1. In the `resin-boot` partition of your SD card, modify the file `config.txt` with the following value: 
```
start_x=1
gpu_mem=128
```

2. Enable the **bcm2835-v4l2** kernel module:

For that, you need a shell access to the resin-os system (not a shell inside a container): You can use the debug SSH access described [here](https://home-assistant.io/developers/hassio/debugging/#ssh-access-to-the-host). Then, once logged, type the following commands:

```
$> mount -o remount,rw  /dev/mmcblk0p2 /
$> echo -e "# Load the bcm2835-v4l2 module \nbcm2835-v4l2" > /etc/modules-load.d/bcm2835-v4l2.conf
```

The first command will remount the main file system with write permission and the second one create a configuration file that will load the kernel module **bcm2835-v4l2** at startup.

```
$> reboot
```

Reboot in order to apply changes.

Now the camera should be exposed in `/dev/`. If you want to check, log in again and type `ls /dev/video*`. For me, as I have just the Raspberry camera, I only see one line:`/dev/video0`

Finally to use the camera in Home Assistant you can use the [ffmpeg component](https://home-assistant.io/components/camera.ffmpeg/). Put something like this inside your `configuration.yaml`:
```(yaml)
ffmpeg:

camera:
  - platform: ffmpeg
    input: /dev/video0
```

> Note: For me, even after this modification the Raspberry camera **component** doesn't work. When I try using the `raspistill` command inside the Home Assistant container, it always give me the same error:
> ```
> mmal: mmal_component_create_core: could not find component 'vc.camera_info'
> mmal: Failed to create camera_info component
> ...
> ```

Done on a Raspberry 2 with a fresh install of Hassio (build 1.3).
Hassio 0.98, Home assistant 0.65.6.

**Comment by robmarkcole** [comment]
@frenck did you ever receive your pi cameras?

**Comment by Dmurph24** [comment]
@robmarkcole do we know of any update on this yet?

**Comment by Chamondgl** [comment]
I'm having the same issue. Is there any way to fix this in hass.io? [This ](https://www.home-assistant.io/components/camera.rpi_camera/)setup didn't work and searching around wasn't helpfull.

**Comment by dragerot** [comment]
> sudo adduser homeassistant video

Solved it....

**Comment by Morphy99** [comment]
> In the resin-boot partition of your SD card, modify the file config.txt with the following value:
start_x=1
gpu_mem=128

My RPi 3 refuses to boot with this??

**Comment by lukie80** [comment]
@Vincent-Ma
If I'm not mistaken, this solution does not work with the current suggested version. It bases on HassOS and does not offer any Linux commands. Looks like the config.txt modification isn't valid any more for HassOS. We have to wait for a fix, I assume.

**Comment by lustigerluke** [comment]
I will leave a comment here, as i got a workaround working for me. Hopefully it helps someone ;)

- I am running home-assistant in Docker on raspberry pi as described in here: https://www.home-assistant.io/docs/installation/docker/ . 
- There is another very comfortable container for streaming raspi cam here: https://datarhei.github.io/restreamer/docs/guides-raspicam.html
-  adding following lines to your config will give you the steam in HomeAssistant:
```
camera:
  - platform: generic
    name: remoteRPiCam
    still_image_url: http://IP.of.Ra.Pi:8080/images/live.jpg
```

**Comment by nirkons** [comment]
> I will leave a comment here, as i got a workaround working for me. Hopefully it helps someone ;)
> 
> * I am running home-assistant in Docker on raspberry pi as described in here: https://www.home-assistant.io/docs/installation/docker/ .
> * There is another very comfortable container for streaming raspi cam here: https://datarhei.github.io/restreamer/docs/guides-raspicam.html
> * adding following lines to your config will give you the steam in HomeAssistant:
> 
> ```
> camera:
>   - platform: generic
>     name: remoteRPiCam
>     still_image_url: http://IP.of.Ra.Pi:8080/images/live.jpg
> ```

This may work on Homeassistant but not on HASSOS as trying to run "raspi-config" doesn't work.

is there really no solution for this?? I'm getting "idle" status and can't see an image using raspberry pi camera

**Comment by internet-conception** [comment]
same issue :( a little bit disapointed... I switch to hass.io to escape this kind of troubles and keep concentrate on designing a cool gui

**Comment by internet-conception** [comment]
Finaly I installed an old usb webcam as replacement and FYI I installed this Repositories : https://www.github.com/HerrHofrat/hassio-addons (motion packet)

So there is a way to add a webcam via a Repositories ;o)

**Comment by ezar** [comment]
Please can you resolve this issue?
Regards,

**Comment by GrowFOUR** [comment]
I found this "Known issues and limitations"
Local USB or Pi cameras are currently not supported yet. This is caused by a limitation of Hass.io. A request for a change has been made to make this possible in the future. motionEye(OS) allows you to use a second instance (e.g., a Pi Zero W) for those kinds of purposes.

That was found here:
https://github.com/hassio-addons/addon-motioneye#known-issues-and-limitations

**Comment by PKHarsimran** [comment]
Did we get this problem fixed yet ?

**Comment by marcinbauer85** [comment]
I would also ask for this, since in 0.110 the rpi_camera has been modified, whats the point if it's not working?

**Comment by Pavel-Efimov** [comment]
any chance to make Raspberry Cam work on hassio?

**Comment by cortado** [comment]
Any resolution? Have RPi3B+ and running latest everything (as of 22.07.2020) and added an RPi cam for surveillance, but cannot get it enabled

**Comment by marcinbauer85** [comment]
https://raspberrypi.stackexchange.com/questions/46851/raspberry-pi-2-camera-and-alpinelinux/51440#51440
Apparently this is the only solution, but I don't know how to do it on a RPI running HassOS.

**Comment by morrowsend** [comment]
I swear the more I try to do the most basic simple things with HassIO the more I realize it's a joke. Every step of the way has been like pulling teeth.

**Comment by lpt2007** [comment]
Any one find solution? I can't enable camera.

If I put start_x=1 in config.txt rpi2 with latest HassOS Release-4 build 20 won't start.

Anyone find solution?

**Comment by gdampf** [comment]
To be honest, HassOS as platform is not really good for low level hardware at all. This is a bit the disadvantage of dockerization.
I solved the camera issue for me in 2018 by installing hass on a vanilla raspian, which I kept up to date by standard apt-process and just recently tweaked by a python 3.8.6 installation and everything is running pretty well. If HassOS would remain the last and only platform for home assistant, I would look for a complete different solution. 
So, I hope, the ability to install hass by hand will remain for the next decade.
Even, if the "one image to run" approach may satisfy the needs of complete newbies, it is much to inflexible. And I don't believe, that this little advantage is worth the effort of maintaining a complete linux-distribution at all.
Back to the roots!

BR
Guido
